test/controllers/declaration/DeclarationHolderRequiredControllerSpec.scala

```diff
@@ -72,83 +75,127 @@ class DeclarationHolderRequiredControllerSpec extends ControllerSpec with Option
         }
       }
+    }
+
+    def verify200(declaration: ExportsDeclaration): Unit =
+      "return 200 (OK)" in {
+        withNewCaching(declaration)
+
+        val result = controller.displayPage(Mode.Normal)(getRequest())
+
+        status(result) mustBe OK
+        verifyPageInvoked()
+      }
 
-      "return 303 (SEE_OTHER)" when {
-        "display page method (GET) is invoked and cache contains already one or more authorisations" in {
-          val declarationHolder: DeclarationHolder = DeclarationHolder(Some("ACE"), Some(Eori("GB56523343784324")), Some(EoriSource.OtherEori))
-          withNewCaching(aDeclarationAfter(request.cacheModel, withDeclarationHolders(declarationHolder)))
+    def verify303(declaration: ExportsDeclaration): Unit =
+      "the declaration contains already one or more authorisations" should {
+        "redirect to the /authorisations-required page" in {
+          val declarationHolder = DeclarationHolder(Some("ACE"), Some(Eori("GB56523343784324")), Some(EoriSource.OtherEori))
+          withNewCaching(aDeclarationAfter(declaration, withDeclarationHolders(declarationHolder)))
 
           val result = controller.displayPage(Mode.Normal)(getRequest())
 
           await(result) mustBe aRedirectToTheNextPage
           thePageNavigatedTo mustBe DeclarationHolderSummaryController.displayPage(Mode.Normal)
         }
       }
-    }
 
-    onJourney(CLEARANCE, OCCASIONAL, STANDARD, SUPPLEMENTARY) { request =>
-      "return 400 (BAD_REQUEST)" when {
-        "the user submits the page but does not answer with yes or no" in {
-          withNewCaching(request.cacheModel)
+  }
 
-          val requestBody = Seq("yesNo" -> "")
-          val result = controller.submitForm(Mode.Normal)(postRequestAsFormUrlEncoded(requestBody: _*))
+  "DeclarationHolderRequiredController.submitForm" when {
 
-          status(result) mustBe BAD_REQUEST
-          verifyPageInvoked()
+    "additional declaration type is STANDARD_PRE_LODGED and" when {
+      List(Choice1040, ChoiceOthers) foreach { choice =>
+        s"AuthorisationProcedureCodeChoice is '${choice.value}' and" when {
+          val declaration = withRequest(STANDARD_PRE_LODGED, withAuthorisationProcedureCodeChoice(choice)).cacheModel
+
+          verify303OnYes(declaration)
+
+          verify303OnNo(declaration)
+
+          verify400(declaration)
         }
       }
+    }
+
+    onJourney(CLEARANCE, OCCASIONAL) { request =>
+      verify303OnYes(request.cacheModel)
 
-      "return 303 (SEE_OTHER)" when {
-        "the user submits the page answering Yes" in {
-          withNewCaching(request.cacheModel)
+      verify303OnNo(request.cacheModel)
 
-          val requestBody = Seq("yesNo" -> "Yes")
-          val result = controller.submitForm(Mode.Normal)(postRequestAsFormUrlEncoded(requestBody: _*))
+      verify400(request.cacheModel)
+    }
+
+    def verify303OnYes(declaration: ExportsDeclaration): Unit =
```

Comment:
Nit: Could you rename this to `verify303ReturnedOnYes` as the '303On' reads like '3030n' that confused me on skim reading ;)